pkgs/development/compilers/dmd/default.nix

```diff
@@ -60,24 +60,46 @@ stdenv.mkDerivation rec {
   # Not using patches option to make it easy to patch, for example, dmd and
   # Phobos at same time if that's required
   patchPhase =
-  lib.optionalString (builtins.compareVersions version "2.092.1" <= 0) ''
+
+  # Migrates D1-style operator overloads in DMD source, to allow building with
+  # a newer DMD
+  lib.optionalString (builtins.compareVersions version "2.088.0" < 0) ''
```

Comment:
there's `lib.versionAtLeast` and `lib.versionOlder` in nixpkgs so use those if possible

https://nixos.org/manual/nixpkgs/stable/#sec-functions-library-strings

Reply:
Done